xbmc/addons/kodi-dev-kit/include/kodi/Filesystem.h

```diff
@@ -2014,24 +2014,51 @@ class ATTRIBUTE_HIDDEN CFile
   /// @param[out] line The buffer to store the data in.
   /// @return True when a line was read, false otherwise.
   ///
+  ///
+  /// ------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// #include <kodi/Filesystem.h>
+  ///
+  /// ...
+  ///
+  /// /* Create the needed file handle class */
+  /// kodi::vfs::CFile myFile;
+  ///
+  /// /* Open the wanted file */
+  /// if (myFile.OpenFile(kodi::GetBaseUserPath("/myFile.txt")))
+  /// {
+  ///   /* Read all lines inside file */
+  ///   while (1)
+  ///   {
+  ///     std::string line;
+  ///     if (!myFile.ReadLine(line))
+  ///       break;
+  ///     fprintf(stderr, "%s\n", line.c_str());
+  ///   }
+  /// }
+  ///
+  /// ~~~~~~~~~~~~~
+  ///
   bool ReadLine(std::string& line)
   {
     using namespace kodi::addon;
 
     line.clear();
     if (!m_file)
       return false;
-    // TODO: Read 1024 chars into buffer. If file position advanced that many
-    // chars, we didn't hit a newline. Otherwise, if file position is 1 or 2
-    // past the number of bytes read, we read (and skipped) a newline sequence.
-    char buffer[1025];
-    if (CAddonBase::m_interface->toKodi->kodi_filesystem->read_file_string(
-            CAddonBase::m_interface->toKodi->kodiBase, m_file, buffer, sizeof(buffer)))
+
+    const auto toKodi = CAddonBase::m_interface->toKodi;
+
+    char buffer[1024];
+    buffer[sizeof(buffer) - 1] = 0; // Force to have always 0 at end
+    while (toKodi->kodi_filesystem->read_file_string(toKodi->kodiBase, m_file, buffer,
+                                                     sizeof(buffer) - 2))
```

Comment:
Off-by-one error: It looks like this should be `sizeof(buffer) - 1`

Reply:
Why not use `strlen()` instead?

Reply:
> Off-by-one error: It looks like this should be `sizeof(buffer) - 1`

Was about discussion on slack with @phunkyfish, to have always for any case an 0 on end of buffer.

> Why not use `strlen()` instead?

Not understand what you mean? `strlen()` makes a search of 0 inside string, but the length becomes known after the call, and to get the size before can not work, as we not know whats in array.

Most correct should be to make as:
```cpp
    char buffer[1024];
    memset(buffer, 0, sizeof(buffer));
```

But to prevent the additional normally not needed work, I used only the 0 set on end. Normally should `read_file_string` always return `true` if the string is set.

Reply:
Ok, makes sense on strlen.

Reply:
> Was about discussion on slack with @phunkyfish, to have always for any case an 0 on end of buffer.

Our API for `read_file_string()` shouldn't put a 0 past the end of a buffer, or it's a buffer overwrite. Instead of defending against API errors I'd make sure the API is correct and use it as intended.